test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/supportingevidence/SupportingEvidenceControllerSpec.scala

```diff
@@ -0,0 +1,1328 @@
+    }
+
+    "display supporting do you want to upload supporting evidence page" when {
+
+      "the use has never answered this question" in {
+        def performAction(): Future[Result] = controller.checkYourAnswers()(FakeRequest())
+
+        val answers = IncompleteSupportingEvidenceAnswers.empty
+
+        val (session, _, _) = sessionWithSingleDisposalState(Some(answers))
+
+        inSequence {
+          mockAuthWithNoRetrievals()
+          mockGetSession(session)
+        }
+
+        checkIsRedirect(
+          performAction(),
+          routes.SupportingEvidenceController.doYouWantToUploadSupportingDocuments()
+        )
+      }
+    }
+
+    "display supporting evidence expired page" when {
+
+      "there are expired supporting evidences" in {
+        def performAction(): Future[Result] = controller.checkYourAnswers()(FakeRequest())
+
+        val supportingEvidence = sample[SupportingEvidence]
+        val answers = IncompleteSupportingEvidenceAnswers.empty.copy(
+          doYouWantToUploadSupportingEvidence = Some(false),
+          List(supportingEvidence),
+          List(supportingEvidence)
+        )
+
+        val (session, _, _) = sessionWithSingleDisposalState(Some(answers))
+
+        inSequence {
+          mockAuthWithNoRetrievals()
+          mockGetSession(session)
+        }
+
+        checkIsRedirect(
+          performAction(),
+          routes.SupportingEvidenceController.supportingEvidenceExpired()
+        )
+      }
+    }
+
+    "display the check your answers page" when {
+
+      "has completed the supporting evidence section" in {
+        def performAction(): Future[Result] = controller.checkYourAnswers()(FakeRequest())
+
+        val answers = CompleteSupportingEvidenceAnswers(
+          doYouWantToUploadSupportingEvidence = false,
+          List.empty
+        )
+
+        val (session, _, _) = sessionWithSingleDisposalState(Some(answers))
+
+        inSequence {
+          mockAuthWithNoRetrievals()
+          mockGetSession(session)
+        }
+
+        checkPageIsDisplayed(
+          performAction(),
+          messageFromMessageKey("supporting-evidence.check-your-answers.title")
+        )
+      }
+
+      "answered the do you want to upload a file question" in {
+        def performAction(): Future[Result] = controller.checkYourAnswers()(FakeRequest())
+
+        val answers = IncompleteSupportingEvidenceAnswers.empty.copy(
+          doYouWantToUploadSupportingEvidence = Some(false),
+          List.empty,
+          List.empty
+        )
+
+        val (session, _, _) = sessionWithSingleDisposalState(Some(answers))
+
+        inSequence {
+          mockAuthWithNoRetrievals()
+          mockGetSession(session)
+        }
+
+        checkPageIsDisplayed(
+          performAction(),
+          messageFromMessageKey("supporting-evidence.check-your-answers.title")
+        )
+      }
+    }
+
+    def performAction(): Future[Result] = controller.checkYourAnswersSubmit()(FakeRequest())
```

Comment:
this could go in it's own section outside of the "display check your answers". Basically, it'd be handy if when we collapse the spec in an editor and when we expand it bit by bit we can see the tests for each endpoint in the controller in separate sections